Parsing YAML fails if it uses type property based polymorphism

handle_request API sometimes returns unexpected YAML content in `choices[0].message.tool_calls.function.arguments` about polymorphism. The client expect the type as a YAML tag like:

```json
{
  "id": "gen-1756382683-fKl2s2jVlj9VESv7zJfw",
  "provider": "Google",
  "model": "anthropic/claude-sonnet-4",
  "object": "chat.completion",
  "created": 1756382686,
  "choices": [
    {
      "logprobs": null,
      "finish_reason": "tool_calls",
      "native_finish_reason": "tool_calls",
      "index": 0,
      "message": {
        "role": "assistant",
        "content": "I need to continue with the architecture setup. Let me fix the app state creation and then set up the screens properly.",
        "refusal": null,
        "reasoning": null,
        "tool_calls": [
          {
            "id": "toolu_vrtx_016X18uDXCcAtnTMvrVDtn4L",
            "index": 0,
            "type": "function",
            "function": {
              "name": "add_app_state",
              "arguments": "{\"appStateYaml\": \"!<CustomDataTypeListAppState>\\nname: \\\"monthsList\\\"\\ndataTypeId: \\\"a52fd49b-6542-40b2-942e-631b14ecf08b\\\"\\ndefaultValue: []\\nuserWritable: true\"}"
            }
          }
        ]
      }
    }
  ],
  "usage": {
    "prompt_tokens": 40422,
    "completion_tokens": 140,
    "total_tokens": 40562
  }
}
```

However, sometimes the endpoint set the type at type property like:

```
{
  "id": "gen-1756382677-YZ6n77BnLaXCJIrpjjRL",
  "provider": "Google",
  "model": "anthropic/claude-sonnet-4",
  "object": "chat.completion",
  "created": 1756382678,
  "choices": [
    {
      "logprobs": null,
      "finish_reason": "tool_calls",
      "native_finish_reason": "tool_calls",
      "index": 0,
      "message": {
        "role": "assistant",
        "content": "I can see from the previous tool execution results that we have successfully created a Month data type, but we need to continue setting up the project architecture. Let me proceed with the next steps:\n\nTool call 1: I'll create the app state for the months list using the correct data type ID",
        "refusal": null,
        "reasoning": null,
        "tool_calls": [
          {
            "id": "toolu_vrtx_01AnHdVPebnhjB9CQRmfcFBi",
            "index": 0,
            "type": "function",
            "function": {
              "name": "add_app_state",
              "arguments": "{\"appStateYaml\": \"name: \\\"monthsList\\\"\\ntype: !<CustomDataTypeListAppState>\\ndataTypeId: \\\"a52fd49b-6542-40b2-942e-631b14ecf08b\\\"\\ndefaultValue: []\\nuserWritable: true\"}"
            }
          }
        ]
      }
    }
  ],
  "usage": {
    "prompt_tokens": 40227,
    "completion_tokens": 178,
    "total_tokens": 40405
  }
}
```

We have fallback to type based deserialization, but it looks like the falback doesn't work now.
https://github.com/ComposeFlow/ComposeFlow/blob/2846e86c339fdf2ed905a4d08bc383186f9020a4/core/model/src/commonMain/kotlin/io/composeflow/serializer/YamlSerializer.kt#L62-L75

Example error message:

```
Caused by: InvalidPropertyValueException at fields[0].fieldType on line 6, column 7: Value for 'fieldType' is invalid: Value is missing a type tag (eg. !<type>)
```

--- a/core/model/src/commonMain/kotlin/io/composeflow/serializer/YamlSerializer.kt
+++ b/core/model/src/commonMain/kotlin/io/composeflow/serializer/YamlSerializer.kt
@@ -58,7 +58,16 @@ val yamlPropertyBasedSerializer =
  * Checks if this throwable or any of its causes is a [MissingTypeTagException].
  */
 val Throwable.isCausedByMissingTypeTag: Boolean
-    get() = this is MissingTypeTagException || this.cause?.isCausedByMissingTypeTag == true
+    get() {
+        var current: Throwable? = this
+        while (current != null) {
+            if (current is MissingTypeTagException) {
+                return true
+            }
+            current = current.cause
+        }
+        return false
+    }
 
 /**
  * Wrapper method that attempts to decode using yamlDefaultSerializer first,

